src/App.tsx

```diff
@@ -15,71 +16,98 @@ export const goods = [
+  handleButtonClick = (good?: string) => {
+    this.setState({
+      selectedGood: good || '',
+    });
+  };
+
+  createGoodsHeader = (selectedGood: string) => {
+    return selectedGood
+      ? (
+        <h1 className="title is-flex is-align-items-center">
+          {`${selectedGood} is selected`}
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => this.handleButtonClick()}
+          />
+        </h1>
+      )
+      : (
+        <h1 className="title">
+          No goods selected
+        </h1>
+      );
+  };
+
+  createGoodButton = (isSelected: boolean, good: string) => {
+    return (
       <button
-        data-cy="ClearButton"
+        data-cy={isSelected ? 'RemoveButton' : 'AddButton'}
         type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        className={
+          classNames('button', {
+            'is-info': isSelected,
+          })
+        }
+        onClick={
+          isSelected
+            ? () => this.handleButtonClick()
+            : () => this.handleButtonClick(good)
+        }
+      >
+        {isSelected ? '-' : '+'}
+      </button>
+    );
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        {this.createGoodsHeader(selectedGood)}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              const isSelected = good === selectedGood;
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              return (
+                <tr
+                  data-cy="Good"
+                  className={
+                    classNames('', {
+                      'has-background-success-light': isSelected,
+                    })
+                  }
+                  key={(new Date().getTime()) * Math.random()}
```

Comment:
you have dood for it
```suggestion
                  key={good}
```

Reply:
Why do you ignore it?